bftengine/src/bcstatetransfer/Messages.hpp

```diff
@@ -64,11 +65,28 @@ struct CheckpointSummaryMsg : public BCStateTranBaseMsg {
   STDigest digestOfResPagesDescriptor;
   uint64_t requestMsgSeqNum;
 
-  static bool equivalent(const CheckpointSummaryMsg* a,
-                         const CheckpointSummaryMsg* b) {
+  static bool equivalent(const CheckpointSummaryMsg* a, const CheckpointSummaryMsg* b) {
     return (memcmp(a, b, sizeof(CheckpointSummaryMsg)) == 0);
   }
 
+  static bool equivalent(const CheckpointSummaryMsg* a, uint16_t a_id,
+                         const CheckpointSummaryMsg* b, uint16_t b_id) {
+    if (memcmp(a, b, sizeof(CheckpointSummaryMsg)) != 0) {
+      auto logger = concordlogger::Log::getLogger("state-transfer");
+      LOG_INFO(logger, "Mismatched Checkpoints for checkpointNum=" << a->checkpointNum  << std::endl

```

Comment:
This will help debugging a lot but I think the log level should be either warning or error.

Reply:
I'll change it to Warning